### PR TITLE
Show a lobby screen in video rooms

### PR DESCRIPTION
**Pull request by robintown**
Depends on https://github.com/vector-im/element-web/pull/21746

Light mode|Dark mode
-|-
Screenshot 2022-04-13 at 11-30-26 Element testingScreenshot 2022-04-13 at 11-30-38 Element testing|Screenshot 2022-04-13 at 11-41-22 Element testingScreenshot 2022-04-13 at 11-41-29 Element testing

Type: enhancement

---
Here's what your changelog entry will look like:

## ✨ Features
 * Show a lobby screen in video rooms ([\#8287](https://github.com/matrix-org/matrix-react-sdk/pull/8287)).

Preview: https://pr8287--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: res/css/views/voip/_VideoLobby.scss
Comment (turt2live) [inline res/css/views/voip/_VideoLobby.scss:19]: we should be using `$spacing` for new code (applies throughout)

File: res/css/views/voip/_VideoLobby.scss
Comment (turt2live) [inline res/css/views/voip/_VideoLobby.scss:47]: Whenever we use `!important` we should be documenting what exactly we're overriding.

File: src/components/views/voip/VideoLobby.tsx
Comment (turt2live) [inline src/components/views/voip/VideoLobby.tsx:27]:
```suggestion
    IconizedContextMenuOption, 
    IconizedContextMenuOptionList,
```

File: src/components/views/voip/VideoLobby.tsx
Comment (turt2live) [inline src/components/views/voip/VideoLobby.tsx:132]: this could do with a comment to explain why we're sending it into an async closure

File: src/stores/VideoChannelStore.ts
Comment (turt2live) [inline src/stores/VideoChannelStore.ts:50]:
```suggestion
    private static readonly TIMEOUT_MS = 8000;
```

File: src/stores/VideoChannelStore.ts
Comment (turt2live) [inline src/stores/VideoChannelStore.ts:80]: should this be resolved in this PR?

**Review by turt2live — CHANGES_REQUESTED** [review]

File: src/stores/VideoChannelStore.ts
Comment (robintown) [reply src/stores/VideoChannelStore.ts:80]: Ah, right, forgot I put that there

File: src/components/views/voip/VideoLobby.tsx
Comment (robintown) [reply src/components/views/voip/VideoLobby.tsx:132]: hm, seems that this could stand to be split into a `useAsyncMemo` + synchronous `useEffect` anyways

File: src/stores/VideoChannelStore.ts
Comment (turt2live) [inline src/stores/VideoChannelStore.ts:61]: public/private/protected

**Review by turt2live — APPROVED** [review]
thanks :)

**Review by gaelledel — CHANGES_REQUESTED** [review]
Screenshot 2022-04-13 at 14 46 42

- [ ] For some reason we can't see the bottom of the frame even when enlarging the view port. The connect now button sits too close to the edge.

Can these be fixed?

**Review by gaelledel — COMMENTED** [review]
@robintown I'm trying to enter the room in order to test the face pile but it gets blocked on Connect now. Therefore nobody is in the room and I cannot see the face pile before entering in the lobby

**Comment by robintown** [comment]
@gaelledel I'm unable to reproduce. Is your browser somehow blocking the Jitsi iframe, or is it requesting camera permissions from you when you try to connect? If the latter, you need to approve those permissions for it to connect successfully.

(I should probably increase the timeout, so there's more time to accept permissions)

**Comment by robintown** [comment]
Currently blocked on figuring out how to make meet.element.io not crash

File: src/components/views/elements/FacePile.tsx
Comment (turt2live) [inline src/components/views/elements/FacePile.tsx:78]: file should be renamed to `RoomFacePile` to reflect new component name, and update any existing imports.

**Review by turt2live — APPROVED** [review]
otherwise lgtm, assuming we can make meet.element.io not crash (is it a code issue or a load issue we're worried about?)

File: src/components/views/elements/FacePile.tsx
Comment (robintown) [reply src/components/views/elements/FacePile.tsx:78]: FacePile is still there as the default export in that file, and RoomFacePile is just a shortcut for it, so I think this should keep its name

Comment (turt2live) [reply src/components/views/elements/FacePile.tsx:78]: this starts to get very tricky under the proposed code style - can we move `RoomFacePile` out to its own file then?

**Comment by robintown** [comment]
The issue with meet.element.io is not load, but that the Jitsi Meet client somehow crashes on varying things while it's trying to start. Tthis doesn't happen when using my personal Jitsi Meet instance, hence why I am confused and need to investigate

**Comment by robintown** [comment]
I'm at a loss for why meet.element.io is getting wedged (appears to be a race internal to Jitsi Meet), so I've opened a support ticket with ops to update the instance, as that's the only other course of action I can think of.

@gaelledel To unblock this in the meantime, I've sent you an invite to a video room that uses my personal Jitsi Meet instance, which doesn't appear to be affected by this issue.

**Review by gaelledel — APPROVED** [review]
It works and looks great :D

**Pull request merged**
